### PR TITLE
Refactor/mp as lib

### PR DESCRIPTION
**Pull request by joaoritter**
The main thing here: Refactored MoneyPool into its own library. 

This separates concerns nicely, and makes calling mp functions from within Fountain way cleaner.
In consequence, we need to store the "id" in the MoneyPool. To be more precise and avoid the problems we've discussed in the past, the idea of a "mpId" has been revised to a "mpNumber". This makes it more clear that the `mps` mapping key is simply using the mp.number as a key. There should never be a mismatch between mp.number and the key of the mapping.

I also brought back the verb "tap". In general "tapping" a money pool refers to an owner withdrawing `want` token sustainments. Collecting from a money pool refers to withdrawing `water` redistributions

Tests pass.

### CODE REVIEW

File: test/FountainV1.test.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
-const Fountain = artifacts.require("FountainV1");
+const Fountain = artifacts.require("Fountain");
```
Comment (tauren) [inline test/FountainV1.test.js:1]: Rename to `Fountain.test.js` removing V1?

File: test-helpers/assertions.js [hunk 2]
```diff
@@ -22,48 +22,12 @@ exports.assertSustainabilityTarget = async (
...
   const currentTarget = (
     await instance.getMp(
-      (await instance.latestMpIds(address)).toNumber()
+      (await instance.latestMpNumber(address)).toNumber()
     )
   ).target.toNumber();
   assert.equal(currentTarget, target, message);
 };
 
-exports.assertInitializeMoneyPoolEvent = async (
```
Comment (tauren) [inline test-helpers/assertions.js:31]: Not testing events anymore?

File: contracts/MoneyPoolOwner.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    /** 
+        @notice This allows your contract to accept sustainments. 
+        @dev You can charge your customers however you like, and they'll keep the surplus if there is any.
+        @param _amount The amount you are taking. Your contract must give Fountain allowance.
+        @param _sustainer Your contracts end user who is sustaining you.
+        Any surplus from your Money pool will be redistributed to this address.
+        @return mpNumber The number of the Money pool that was sustained.
+    */
+    function _sustainMp(uint256 _amount, address _sustainer)
```
Comment (tauren) [inline contracts/MoneyPoolOwner.sol:93]: Just want to clarify, does this let other contracts can sustain Fountain? The comments make it sound like it is required, how is that enforced?

**Review by tauren — APPROVED** [review]
Solid refactor. Definitely like the separation of concerns and lightening the size of various files. Feels like we should be able to do more of this, but not sure yet where...

File: test/FountainV1.test.js (hunk 1 above)
Comment (joaoritter) [reply test/FountainV1.test.js:1]: nice

File: test-helpers/assertions.js (hunk 2 above)
Comment (joaoritter) [reply test-helpers/assertions.js:31]: i removed the initialize event and the activate event. Those were both events i added unsure of how the front-end would need them. Turns out we're good to go without them.

File: contracts/MoneyPoolOwner.sol (hunk 3 above)
Comment (joaoritter) [reply contracts/MoneyPoolOwner.sol:93]: This makes it so a contract's users can sustain the contract's MP. We haven't built it dependencies yet

**Comment by joaoritter** [comment]
rebased!

**Pull request merged**